packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -753,6 +753,43 @@ export class ForkChoice implements IForkChoice {
     return;
   }
 
+  /**
+   * Find dependent root of a head block
+   * epoch - 2 | epoch - 1 | epoch
+   * pivotSlot |     -     | blockSlot
+   */
+  findAttesterDependentRoot(headBlockHash: Root): RootHex | null {
```

Comment:
Should this have an epoch argument? I guess this `headBlockHash` comes from a checkpoint so the `headBlockHash` could be from older epochs. In that case you need the epoch argument to now how back you have to go.

Reply:
no we don't need that, it comes from the attestation's `beaconBlockRoot`, not the checkpoint

```typescript
const pivotBlockRoot = forkChoice.findAttesterDependentRoot(data.beaconBlockRoot);
```

Reply:
If we use the target root from the attestation we could save some hops right?

Reply:
@dapplion  1st hop was already the target root from the attestation

```typescript
// 1st hop: target block
block = this.getBlockHex(block.targetRoot);
```